### ISSUE TITLE
SSO support for exchangecalendar 

### ISSUE THREAD

**Issue by begkatten** [Ericsson/exchangecalendar#129]
Hi,

Some users use SSO and exchangecalendar continues to ask for passwd. Couldn't the krb5 ticket be used for authentication for the calendar too?

Cheers,
/István

**Comment by bavincen** [comment]
Hi @begkatten 
I tried to login in thunderbird setting connection security to "STARTTLS" and Authentication method to "Kerbros/GSSAPI",  it created account but when i click Get Messages  notification popsup with Error 

`The Kerberos/GSSAPI ticket was not accepted by the POP server. Please check that you are logged in to the Kerberos/GSSAPI realm. `

How do I fix this
Does this error saying should login in operating system somewhere ( iam using Ubuntu)

**Comment by xnandersson** [comment]
Try to see first if GSSAPI is one of the possible authentication 
mechanisms in the pop-server.

You try this out by doing a telnet to the port of the POP-server, where 
you then write

EHLO localhost.

I provide sample output when I connect to my server. You can see I only 
support two auth-mechanisms (one really) AUTH PLAIN LOGIN and AUTH=PLAIN 
LOGIN. I recall that the second one is there as an ugly work-around for 
some type of mailservers that didn't interpret the RFC's the correct way.

nandersson@nandersson-ThinkPad-X1-Carbon-2nd:~$ telnet smtp.openforce.se 587
Trying 74.50.52.122...
Connected to ns2.openforce.se.
Escape character is '^]'.
220 ns2.openforce.se ESMTP Postfix (Ubuntu)
*EHLO localhost*
250-ns2.openforce.se
250-PIPELINING
250-SIZE 10240000
250-VRFY
250-ETRN
250-STARTTLS
*250-AUTH PLAIN LOGIN**
**250-AUTH=PLAIN LOGIN*
250-ENHANCEDSTATUSCODES
250-8BITMIME
250 DSN

Hope this helps.

Best regards,
Niklas

**Comment by bavincen** [comment]
Hello Niklas 
Thanks a lot for this info. btw  i am using IMAP.. will that work?

Error appeared when i typed in cmd
`
Escape character is '^]'.
* OK The Microsoft Exchange IMAP4 service is ready.
EHLO localhost
EHLO BAD Command Error. 12
`

**Comment by xnandersson** [comment]
Ah, sorry about that. Here is a link [1] that should explain how to 
debug a IMAP-connection using telnet.

This is the interesting section:

Escape character is '^]'.
* OK [CAPABILITY IMAP4rev1 UIDPLUS CHILDREN NAMESPACE THREAD=ORDEREDSUBJECT THRE
AD=REFERENCES SORT QUOTA IDLE*AUTH=CRAM-MD5*  SMAP1 KEYWORDS ACL ACL2=UNION STARTT
LS XMAGICTRASH] Courier-IMAP ready. Copyright 1998-2008 Double Precision, Inc.
See COPYING for distribution information.


[1] http://www.courier-mta.org/imap/tutorial.setup.html

Regards,
Niklas

**Comment by bavincen** [comment]
Niklas Thanks again. I think you didnt see I am able to type `EHLO localhost`
error is like `EHLO BAD Command Error`

**Comment by xnandersson** [comment]
Yes, sorry about that. It was my fault. I mixed up SMTP and IMAP. It is 
in SMTP that you use the EHLO (used to be HELO). In IMAP you should get 
the output per the link I send. Seems like you shouldn't need to write 
any specific commands to list the server capabilities.

**Comment by xnandersson** [comment]
...when I test my server I get:

nandersson@nandersson-ThinkPad-X1-Carbon-2nd:~$ telnet ns1.openforce.se 143
Trying 65.99.241.185...
Connected to ns1.openforce.se.
Escape character is '^]'.
* OK [CAPABILITY IMAP4rev1 LITERAL+ SASL-IR LOGIN-REFERRALS ID ENABLE 
IDLE STARTTLS LOGINDISABLED] Dovecot (Ubuntu) ready.

...if I had support for GSSAPI I suppose that support would show there.

BUT

...it might be that Microsoft has masked it out in the server boiler 
plate, as it is sometimes considered good security practise. I.e you 
don't want to make it easy for an intruder to see your exact server 
capabilities, in order for them to do a more targeted attack.

Regards,
Niklas

**Comment by bavincen** [comment]
`250-SIZE 13631488
250-PIPELINING
250-DSN
250-ENHANCEDSTATUSCODES
250-AUTH
250-8BITMIME
250-BINARYMIME
250-CHUNKING
250-XEXCH50
250 XSHADOW
451 4.7.0 Timeout waiting for client input`


Got this :)
